### PR TITLE
Refactor API handlers and implement storage queries

### PR DESCRIPTION
**Pull request by AmineAfia**
# API Enhancements and Storage Integration

## Changes
- Renamed `Aggregate` to `Aggregates` in `QueryParams` struct
- Updated route handler names for better clarity:
  - `GetTransactionsWithContract` to `GetTransactionsByContract`
  - `GetEventsWithContract` to `GetLogsByContract`
  - `GetTransactionsWithContractAndSignature` to `GetTransactionsByContractAndSignature`
  - `GetEventsWithContractAndSignature` to `GetLogsByContractAndSignature`
- Implemented `handleLogsRequest` and `handleTransactionsRequest` functions to centralize logic
- Integrated storage layer with API handlers:
  - Added `getMainStorage` function to create storage connector
  - Updated `GetLogs` and `GetTransactions` methods to use storage queries
- Enhanced `QueryFilter` struct with additional fields for filtering and aggregation
- Modified `ClickHouseConnector` methods to remove block number filtering and prepare for more flexible querying
- Renamed `newConnector` to `NewConnector` and made it public

## TODO

- Implement total items count and total pages count in API responses
- Add aggregations, filters, and grouping to log queries in ClickHouse connector

### CODE REVIEW

**Comment by AmineAfia** [comment]
* **#43** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/43?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @AmineAfia and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: internal/handlers/logs_handlers.go [hunk 1]
```diff
@@ -7,115 +7,94 @@ import (
...
-			"sum_value":     "1000000000000000000000",
-			"avg_gas_price": "20000000000",
+			"aggregates": queryParams.Aggregates,
 		},
 	}
 
+	sendJSONResponse(w, response)
+}
+
+func getMainStorage() (storage.IMainStorage, error) {
```
Comment (iuwqyir) [inline internal/handlers/logs_handlers.go:84]: Do you want to create a new client for every endpoint or should it be 1 for the API?

Comment (AmineAfia) [reply internal/handlers/logs_handlers.go:93]: done

**Review by iuwqyir — APPROVED** [review]

**Comment by AmineAfia** [comment]
### Merge activity

* **Sep 26, 6:12 AM EDT**: @AmineAfia started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thirdweb-dev/indexer/43).

**Pull request merged**